src/compiler/checker.ts

```diff
@@ -2598,8 +2598,13 @@ namespace ts {
         }
 
         function getAlternativeContainingModules(symbol: Symbol, enclosingDeclaration: Node): Symbol[] {
-            const results: Symbol[] = [];
             const containingFile = getSourceFileOfNode(enclosingDeclaration);
+            const id = "" + getNodeId(containingFile);
+            const links = getSymbolLinks(symbol);
+            if (links.extendedContainersByFile && links.extendedContainersByFile.has(id)) {

```

Comment:
I would avoid the double lookup (has/get) if possible, as I don't believe any engine optimizes this case yet.